File: test/src/test/java/org/camunda/bpm/extension/process_test_coverage/rules/MethodCoverageTest.java
package org.camunda.bpm.extension.process_test_coverage.rules;

import org.camunda.bpm.engine.test.Deployment;
import org.camunda.bpm.extension.process_test_coverage.junit.rules.TestCoverageProcessEngineRule;
import org.camunda.bpm.extension.process_test_coverage.junit.rules.TestCoverageProcessEngineRuleBuilder;
import org.junit.Rule;
import org.junit.Test;

import java.util.HashMap;
import java.util.Map;

import static org.hamcrest.Matchers.*;

/**
 * Test case starting an in-memory database-backed Process Engine.
 */
public class MethodCoverageTest {

	private static final String PROCESS_DEFINITION_KEY = "process-test-coverage";
	
	@Rule
	public TestCoverageProcessEngineRule rule = TestCoverageProcessEngineRuleBuilder.create()
			.withDetailedCoverageLogging().build();

	@Test
	@Deployment(resources = "process.bpmn")
	public void testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered() {
		
	    Map<String, Object> variables = new HashMap<String, Object>();
		variables.put("path", "A");
		rule.getRuntimeService().startProcessInstanceByKey(PROCESS_DEFINITION_KEY, variables);
		
		rule.addTestMethodCoverageAssertionMatcher("testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered", greaterThan(6.9 / 11.0));
	    rule.addTestMethodCoverageAssertionMatcher("testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered", lessThan(7.1 / 11.0));

	}

	@Test
	@Deployment(resources = "process.bpmn")
	public void testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered() {
		Map<String, Object> variables = new HashMap<String, Object>();
		variables.put("path", "B");
		rule.getRuntimeService().startProcessInstanceByKey(PROCESS_DEFINITION_KEY, variables);
		
		rule.addTestMethodCoverageAssertionMatcher("testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered", greaterThan(6.9 / 11.0));
        rule.addTestMethodCoverageAssertionMatcher("testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered", lessThan(7.1 / 11.0));
	}

}

File: extension/junit5-platform-7/src/test/kotlin/org/camunda/bpm/extension/process_test_coverage/junit5/MethodCoverageTest.kt
package org.camunda.bpm.extension.process_test_coverage.junit5

import org.assertj.core.api.HamcrestCondition
import org.camunda.bpm.engine.test.Deployment
import org.hamcrest.Matchers
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.RegisterExtension

class MethodCoverageTest {

    companion object {
        const val PROCESS_DEFINITION_KEY = "process-test-coverage"
        @JvmField
        @RegisterExtension
        var extension: ProcessEngineCoverageExtension = ProcessEngineCoverageExtension.builder().withDetailedCoverageLogging().build()

    }

    @Test
    @Deployment(resources = ["process.bpmn"])
    fun testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered() {
        val variables: MutableMap<String, Any> = HashMap()
        variables["path"] = "A"
        extension.processEngine.runtimeService.startProcessInstanceByKey(PROCESS_DEFINITION_KEY, variables)
        extension.addTestMethodCoverageCondition(
                "testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered",
                HamcrestCondition(Matchers.greaterThan(6.9 / 11.0))
        )
        extension.addTestMethodCoverageCondition(
                "testCoverageWhenRunningPathAShouldReportSevenOutOfElevenElementsCovered",
                HamcrestCondition(Matchers.lessThan(7.1 / 11.0))
        )
    }

    @Test
    @Deployment(resources = ["process.bpmn"])
    fun testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered() {
        val variables: MutableMap<String, Any> = HashMap()
        variables["path"] = "B"
        extension.processEngine.runtimeService.startProcessInstanceByKey(PROCESS_DEFINITION_KEY, variables)
        extension.addTestMethodCoverageCondition(
                "testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered",
                HamcrestCondition(Matchers.greaterThan(6.9 / 11.0))
        )
        extension.addTestMethodCoverageCondition(
                "testCoverageWhenRunningPathBShouldReportSevenOutOfElevenElementsCovered",
                HamcrestCondition(Matchers.lessThan(7.1 / 11.0))
        )
    }


}